Changes for TimescaleDB 2.0.0-rc2 launch (#543)
timescale/docs.timescale.com-content: update-timescaledb/update-tsdb-2.md

# Updating TimescaleDB to 2.0 [](update)

Use these instructions to update TimescaleDB 1.x to TimescaleDB 2.0

>:WARNING:These instructions are only for upgrading TimescaleDB 1.x to TimescaleDB 2.0
> If you need to upgrade your existing TimescaleDB 1.x to a newer version in the 1.x

> > release line (eg. 1.7.2 to 1.7.4), please see Update [TimescaleDB 1.x][update-tsdb-1]

Reviewer: ```suggestion
> release line (eg. 1.7.2 to 1.7.4), please see Update [TimescaleDB 1.x][update-tsdb-1].
```

Author: Oops! I meant to go back through and look at periods in all docs, especially bulleted lists.  Thanks!

### TimescaleDB Release Compatibility

TimescaleDB 2.0 currently supports the following PostgreSQL releases. If you are not currently running 
a compatible release, please upgrade before updating TimescaleDB.

 TimescaleDB Release |   Supported PostgreSQL Release
 --------------------|-------------------------------
 2.0-RC1+            | 11, 12

>:TIP:If you need to upgrade PostgreSQL first, please see [our documentation][upgrade-pg].

### Notice of breaking changes from TimescaleDB 1.3+
TimescaleDB 2.0 supports **in-place updates** just like previous releases. During the update, scripts will automatically configure
updated features to work as expected with TimescaleDB 2.0.

> Because this is our first major version release in two years, however, we’re providing additional guidance to help you ensure the update completes successfully and everything is configured as expected (and optimized for your workload). In particular, settings related to [Continuous Aggregates][caggs], [compression][compression], and [data retention][retention] have been modified to provide greater configuration transparency and flexibility, therefore we highly recommend verifying that these settings were migrated correctly.

Reviewer: Small nit - generally recommend using hard line breaks, so that later diffs can be more easily compared via git at the line (vs. paragraph) level.

**Before completing the upgrade**, we encourage you to read [Changes in TimescaleDB 2.0][changes-in-ts2] for a more detailed look at the major changes in TimescaleDB 2.0 and how they impact the way your applications and scripts interact with the API.

> ### Prerequisites

Reviewer: And anchor tags so we can in-page link?

#### PostgreSQL Compatibility
**TimescaleDB 2.0 is not compatible with PostgreSQL 9.6 or 10**. If your current PostgreSQL installation is not at least version 11, please upgrade PostgreSQL first. Depending on your current PostgreSQL version and installed TimescaleDB release, you may have to perform multiple upgrades because of compatibility restrictions.

For example, if you are currently running PostgreSQL 10 and TimescaleDB 1.5, the recommended upgrade path to PostgreSQL 12 and TimescaleDB 2.0 would be:

1. Update TimescaleDB 1.5 to TimescaleDB 1.7 on PostgreSQL 10
1. Upgrade PostgreSQL 10 to PostgreSQL 12 with TimescaleDB 1.7 installed
1. Update TimescaleDB 1.7 to TimescaleDB 2.0 on PostgreSQL with the instructions below

>:TIP: Whenever possible, prefer the most recent supported version, PostgreSQL 12. Please see our [Upgrading PostgreSQL][upgrade-pg] guide for help.

#### Fix Continuous Aggregate Errors Before Upgrading

> Before starting the upgrade to TimescaleDB 2.0, **we highly recommend checking the database log for errors related to failed retention policies that were occurring in TimescaleDB 1.x** and then either remove them or update them to be compatible with existing continuous aggregates. Any remaining retention policies that are still incompatible with the 'ignore_invalidation_older_than' setting will automatically be disabled during the upgrade and a notice provided.

Reviewer: Backticks for code block formatting

```suggestion
Before starting the upgrade to TimescaleDB 2.0, **we highly recommend checking the database log for errors related to failed retention policies that were occurring in TimescaleDB 1.x** and then either remove them or update them to be compatible with existing continuous aggregates. Any remaining retention policies that are still incompatible with the `ignore_invalidation_older_than` setting will automatically be disabled during the upgrade and a notice provided.
```

> >:TIP:Read more about changes to continuous aggregates and data retension policies [here][retention-cagg-changes]

Reviewer: ```suggestion
>:TIP:Read more about changes to continuous aggregates and data retension policies [here][retention-cagg-changes].
```
